server/embed/config.go

```diff
@@ -660,6 +665,27 @@ func updateCipherSuites(tls *transport.TLSInfo, ss []string) error {
 	return nil
 }
 
+func updateMinMaxVersions(info *transport.TLSInfo, min, max string) error {
+	minVersion, err := tlsutil.GetTLSVersion(min)
+	if err != nil {
+		return err
+	}
+
+	maxVersion, err := tlsutil.GetTLSVersion(max)
+	if err != nil {
+		return err
+	}
+
+	// Check if both min and max were provided and min is greater than max
+	if minVersion != 0 && maxVersion != 0 && minVersion > maxVersion {
```

Comment:
This validation doesn't check if cipher suites are configured. As cipher suits only work with TLS 1.2, we should not allow users to set them if maxTLS is not set to TLS1.2

Reply:
> This validation doesn't check if cipher suites are configured. As cipher suits only work with TLS 1.2, we should not allow users to set them if maxTLS is not set to TLS1.2

You must be able to set cipher suites for TLS 1.2 in case you have both TLS 1.2 and TLS 1.3 enabled (the default), so what you said does not make sense. What you probably wanted is failing validation if min-version and max-version is TLS 1.3.

Reply:
I added a check when TLSv1.3 is the only enabled version. 

It is not out of the question that there will be configurability in the future versions of Go and the check might need to be removed later. At this time, it is of course unknown how the possible configurability would look like.